autoload/lsp/diag.vim

```diff
@@ -404,6 +404,12 @@ def DiagsUpdateLocList(bnr: number): bool
     setbufvar(bnr, 'LspQfId', getloclist(0, {id: 0}).id)
   endif
 
+  # if pre-existing error list, make current
+  if LspQfId != 0
+    var LspQfNr: number = getloclist(0, {id: LspQfId, nr: 0}).nr
+    silent! exe $':{LspQfNr} lhistory'
```

Comment:
This code should be moved to the ShowAllDiags() function.  Otherwise whenever new diagnostics is received from the language server, the current location list will be changed.  If the user is using some other plugin (e.g. grep) to populate the location list, then this will lead to unexpected behavior.  In the ShowAllDiags() function, you can get the 'LspQfId' buffer-local variable for the current buffer and make that the current location list.

Reply:
Nice spot - thanks. I've made the change now.